function_mapper.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
```

Comment:
Also, don't forget to include `string` here. It's not mandatory if you include `store.hpp` as I mentioned in the other comment, but it's more clear that `string` is a dependency for this file.